tracker/tracker-assist/src/ScreenRecordingState.ts

```diff
@@ -0,0 +1,149 @@
+import ConfirmWindow from './ConfirmWindow/ConfirmWindow.js'
+import { recordRequestDefault, } from './ConfirmWindow/defaults.js'
+import type { Options as AssistOptions, } from './Assist'
+
+export enum RecordingState {
+  Off,
+  Requested,
+  Recording,
+}
+
+const defaultStyles = '2px dashed red'
+const leftTop = { left: 0, top: 0, position: 'fixed', }
+const bottomRight = { right: 0, bottom: 0, position: 'fixed', }
+
+const borderEmulationStyles = {
+  left: {
+    ...leftTop,
+    height: '100vh',
+    width: 0,
+    borderLeft: defaultStyles,
+  },
+  top: {
+    ...leftTop,
+    height: 0,
+    width: '100vw',
+    borderTop: defaultStyles,
+  },
+  right: {
+    ...bottomRight,
+    height: '100vh',
+    width: 0,
+    borderRight: defaultStyles,
+  },
+  bottom: {
+    ...bottomRight,
+    height: 0,
+    width: '100vw',
+    borderBottom: defaultStyles,
+  },
+}
+
+export default class ScreenRecordingState {
+  public status = RecordingState.Off
+  private agentsRecordingSession: string[] = []
+  private overlayAdded = false
+
+  constructor(
+    private readonly onAccept: () => void,
+    private readonly onDeny: () => void,
+    private readonly options: AssistOptions
+  ) {}
+
+  public get isActive() {
+    return this.status !== RecordingState.Off
+  }
+
+  private confirm: ConfirmWindow | null = null
+
+  public requestRecording = (id: string) => {
+    if (this.status !== RecordingState.Off) return
+    this.status = RecordingState.Requested
+
+    this.confirm = new ConfirmWindow(recordRequestDefault(this.options.controlConfirm))
+    this.confirm.mount().then(allowed => {
+      if (allowed) {
+        this.acceptRecording()
+        this.agentsRecordingSession.push(id)
+      } else {
+        this.confirm?.remove()
+        this.denyRecording()
+      }
+    })
+    .then(() => {
```

Comment:
you can put it after `.catch` to make behave as `.finally()` (which is less maintained in browsers than webRTC)

Reply:
its a hack that shouldn't be used, plus eslint barks when you try to use it

Reply:
Why not, what's wrong with it?